### PR TITLE
Added type hints

### PR DESCRIPTION
**Pull request by yhl48**
# What does this PR do?
Added type hints as requested in #16059.

@Rocketknight1

### CODE REVIEW

File: src/transformers/models/ibert/modeling_ibert.py [hunk 1]
```diff
@@ -882,17 +883,18 @@ def set_output_embeddings(self, new_embeddings):
...
+        attention_mask: Optional[torch.FloatTensor] = None,
+        token_type_ids: Optional[torch.LongTensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[torch.LongTensor] = None,
+        **kwargs: Optional[Dict[str, Any]],
```
Comment (Rocketknight1) [inline src/transformers/models/ibert/modeling_ibert.py:896]:
```suggestion
        **kwargs,
```
In general, I don't think kwargs gets a type annotation in our repo! We might change this later, but for now we can leave it out.

File: src/transformers/models/ibert/modeling_ibert.py [hunk 2]
```diff
@@ -1281,18 +1283,18 @@ def __init__(self, config):
...
+        attention_mask: Optional[torch.FloatTensor] = None,
+        token_type_ids: Optional[torch.LongTensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        start_positions: Optional[torch.LongTensor] = None,
+        end_positions: Optional[torch.LongTensor] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/ibert/modeling_ibert.py:1296]: The order of arguments has changed in some cases here! (This is true in the other methods too). We'd prefer not to change that - can you double-check that and put them back in the original order?

**Review by Rocketknight1 — COMMENTED** [review]
Hey, looks good, thank you! Can you double-check the argument order issue? After that's resolved, this should be ready to merge!

File: src/transformers/models/ibert/modeling_ibert.py (hunk 2 above)
Comment (yhl48) [reply src/transformers/models/ibert/modeling_ibert.py]: fixed in [c813533](https://github.com/huggingface/transformers/pull/16389/commits/c813533a713b0665e7c97ae71dd59ec160fae7b2)

File: src/transformers/models/ibert/modeling_ibert.py (hunk 1 above)
Comment (yhl48) [reply src/transformers/models/ibert/modeling_ibert.py]: fixed in [c813533](https://github.com/huggingface/transformers/pull/16389/commits/c813533a713b0665e7c97ae71dd59ec160fae7b2), but note that in the doc for this model kwargs does get a type annotation

Comment (Rocketknight1) [reply src/transformers/models/ibert/modeling_ibert.py]: That's a fair point - at some point we'll have to figure out a consistent policy on that.

**Review by Rocketknight1 — APPROVED** [review]
Looks great now, thank you!

**Pull request merged**